internal/provisioning/client.go

```diff
@@ -0,0 +1,46 @@
+//go:generate go run -mod=mod github.com/deepmap/oapi-codegen/cmd/oapi-codegen --config client.cfg.yml provisioning.v1.yml
+
+package provisioning
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+)
+
+type ProvisioningClient struct {
+	url    string
+	client *http.Client
+}
+
+type ProvisioningClientConfig struct {
+	URL string
+}
+
+func NewClient(conf ProvisioningClientConfig) (*ProvisioningClient, error) {
+	pc := ProvisioningClient{
+		url:    conf.URL,
+		client: &http.Client{},
+	}
+
+	return &pc, nil
+}
+
+func (pc *ProvisioningClient) request(method, url string, headers map[string]string, body io.Reader) (*http.Response, error) {
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, err
+	}
+
+	for k, v := range headers {
+		req.Header.Add(k, v)
+	}
+
+	return pc.client.Do(req)
+}
+
+func (pc *ProvisioningClient) GetAccountID(sourceID string, identity string) (*http.Response, error) {
+	return pc.request("GET", fmt.Sprintf("%s/sources/%s/account_identity", pc.url, sourceID), map[string]string{
+		"x-rh-identity": identity,
+	}, nil)
+}
```

Comment:
Is there a reason why you don't pass context.Context into this function so you do not need to care about identity argument. Assuming you put identity into the context at the beginning of the request. Then you can simply do:

    req.Header.Set("X-RH-Identity", identity.GetIdentityHeader(ctx))

In fact, you don't need to do this in every method of this client, you can actually implement a `Doer` single-method interface and decorate the `ProvisioningClient` in the `NewClient` function. After that, you don't need to worry about identity ever again for our app.

We use `Doer` for trace context, however, we still add individual headers in our own codebase, I would like to change that :-)

Reply:
Good point, I tried to just copy what we do for the composer client. But that's much cleaner.